src/codemagic/models/xcodebuild.py

```diff
@@ -256,22 +256,32 @@ def test(self,
         error_message = f'Failed to test {self.workspace or self.project}'
         self._run_command(cmd, error_message)
 
+    def show_build_settings(self):
+        cmd = [*self._construct_base_command(None), '-showBuildSettings']
+        error_message = f'Failed to obtain build settings {self.workspace or self.project}'
+
+        cli_app = self.get_current_cli_app()
+        print_streams = cli_app.verbose if cli_app else True
+
+        # avoid formatting build settings output
+        self._run_command(cmd, error_message, print_streams=print_streams, format_streams=False)
+
     def _run_command(self,
                      command: List[str],
                      error_message: str,
-                     ignore_error_code: Optional[int] = None):
+                     print_streams: bool = True,
+                     format_streams: bool = True):
```

Comment:
I find `format_streams` as argument name to be slightly misleading. In case `XcodeBuild` instance does not have `xcpretty`, then there is no formatting even if `format_streams` is set to `True`. I would rename it to something like `allow_xcpretty_formatting` instead to reflect actual meaning.